src/CommandLine.cc

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 1996-2018 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+
+#include "CommandLine.h"
+
+static void
+ResetGetopt(const bool allowStderrWarnings)
+{
+    opterr = allowStderrWarnings ? 1 : 0;
+    // getopt(3) uses global state but resets it if optind is zero
+    // getopt(3) always skips argv[0], even if optind is zero
+    optind = 0;
+}
+
+CommandLine::CommandLine(int argC, char *argV[], const char *shortRules, const struct option *longRules):
+    argv_(),
+    shortOptions_(xstrdup(shortRules)),
+    longOptions_()
```

Comment:
What is the benefit of explicit constructing argv_() and longOptions() with default constructors?

Reply:
I suspect this is done to indicate that the developer has thought about those member initialization rather than forgotten to initialize them.